File: Simple Color Tracker/for Video/Color.Tracker.py
import cv2
import numpy as np

cap = cv2.VideoCapture("blue-track.mp4")

while True:
    ret, frame = cap.read()
    if not ret:
        break
    hsv = cv2.cvtColor(frame, cv2.COLOR_BGR2HSV)
    
    lower_blue = np.array([100,50,50])
    upper_blue = np.array([130,255,255])
    
    mask = cv2.inRange(hsv, lower_blue, upper_blue)
    contours, _ = cv2.findContours(mask, cv2.RETR_TREE, cv2.CHAIN_APPROX_SIMPLE)
    centers, radius = cv2.minEnclosingCircle(contours[0])
    centers = int(centers[0]), int(centers[1])
    radius = int(radius)
    cv2.circle(frame, centers, radius, (0,0,255), 2)
    cv2.imshow('image',frame)
    
    k = cv2.waitKey(5) & 0xFF
    if k == 27:
        break

cv2.destroyAllWindows()
cap.release()

File: Simple Color Tracker/for Video/ColorTracker.cpp
#include <opencv2/opencv.hpp>

int main() {

  cv::VideoCapture cap("blue-track.mp4");

  while (true) {
    cv::Mat frame;
    cap >> frame;
    if (frame.empty()) {
      break; 
    }

    cv::Mat hsv;
    cv::cvtColor(frame, hsv, cv::COLOR_BGR2HSV);

    cv::Scalar lower_blue(100,50,50); 
    cv::Scalar upper_blue(130,255,255);

    cv::Mat mask;
    cv::inRange(hsv, lower_blue, upper_blue, mask);

    std::vector<std::vector<cv::Point> > contours;
    cv::findContours(mask, contours, cv::RETR_TREE, cv::CHAIN_APPROX_SIMPLE);

    if (contours.size() > 0) {
      cv::Point2f center; 
      float radius;
      cv::minEnclosingCircle(contours[0], center, radius);

      int x = (int)center.x;
      int y = (int)center.y;
      int r = (int)radius;

      cv::circle(frame, cv::Point(x,y), r, cv::Scalar(0,0,255), 2);
    }

    cv::imshow("image", frame);

    char k = (char)cv::waitKey(5);
    if (k == 27) {
      break;
    }
  }

  cv::destroyAllWindows();
  cap.release();

  return 0;
}